### ISSUE TITLE
Can we use take the result of embedding layer as target?

### ISSUE THREAD

**Issue by bojone** [fchollet/keras#3288]
I want to train a model to identify the keyword of a sentences (assumpt just one keyword in a sentence).

My oringal thought is:
1. train word vectors using word2vec
2. train a lstm network using keras. Its input is the matrix (whose row is the word2vec vector of the words of sentence) of sentence, and the target is the word2vec vector of the keyword. Loss is mse or cosine_proximity.
3. if we want to identify the keyword of a new sentences, just rank the cos of the result and every word vector of sentence's words.

This way need to pretrain word vectors. But I know keras provide a embedding layer. Can we only use keras to realize my goal? In other word, Can we use take the result of embedding layer as target? Because in my way, the target is the result of the embedding layer?

**Comment by sileod** [comment]
You can use the target as an *input*, use it to compute the distance and add that distance to the loss using ActivityRegularizer
